frontend/src/framework/internal/components/PersistenceManagementDialog/snapshotManagementContent.tsx

```diff
@@ -0,0 +1,507 @@
+                        placeholder="Search title"
+                        onChange={handleTitleFilterValueChange}
+                        className="h-6"
+                    />
+                </Label>
+                <Label text="Last visited at" wrapperClassName="min-w-2xs">
+                    <DateRangePicker
+                        onChange={handleFilterRangeChange}
+                        className="webviz-eds-date-range-picker --compact rounded focus-within:outline-0 border border-gray-300 h-10"
+                    />
+                </Label>
+            </div>
+            <div className="flex gap-4 mb-2 items-center">
+                <Label text="Show my snapshots only" wrapperClassName="flex items-center" position="right">
+                    <Switch checked={tableFilter.ownerId === userId} onChange={handleShowMySnapshotsOnlyChange} />
+                </Label>
+                <Label text="Hide deleted snapshots" wrapperClassName="flex items-center" position="right">
+                    <Switch checked={tableFilter.snapshotDeleted} onChange={handleHideDeletedSnapshotsChange} />
+                </Label>
+                <span className="grow" />
+                <Tooltip title={"Open selected snapshot"} placement="top" enterDelay="medium">
+                    <Button
+                        color="primary"
+                        disabled={!selectedSnapshotId || selectedSnapshot?.snapshotDeleted}
+                        onClick={handleOpenSnapshotClick}
+                        size="medium"
+                    >
+                        <FileOpen fontSize="inherit" /> Open
+                    </Button>
+                </Tooltip>
+                <Tooltip title={"Delete selected snapshot"} placement="top" enterDelay="medium">
+                    <Button
+                        color="danger"
+                        disabled={!selectedSnapshotId || deletePending || selectedSnapshot?.snapshotDeleted}
+                        onClick={handleDeleteClick}
+                        size="medium"
+                    >
+                        {deletePending ? <CircularProgress size="small" /> : <Delete fontSize="inherit" />} Delete
+                    </Button>
+                </Tooltip>
+                <Tooltip title="Refresh list" placement="top" enterDelay="medium">
+                    <Button color="primary" onClick={handleRefreshClick} size="medium">
+                        {snapshotsQuery.isFetching ? <CircularProgress size="small" /> : <Refresh fontSize="inherit" />}{" "}
+                        Refresh
+                    </Button>
+                </Tooltip>
+            </div>
+            <Table
+                rowIdentifier="snapshotId"
+                alternatingColumnColors={USE_ALTERNATING_COLUMN_COLORS}
+                columns={TABLE_COLUMNS}
+                rows={tableData}
+                numPendingRows={snapshotsQuery.isLoading || snapshotsQuery.isFetchingNextPage ? QUERY_PAGE_SIZE : 0}
```

Comment:
Same issue here as for the sessionManagementContent. If resolved there, apply same changes here.